module-services/service-cellular/Pdu.cpp

```diff
@@ -0,0 +1,972 @@
+        static bool parseContentType(Octets &octets, ContentValue &value)
+        {
+            return parseGeneric<DefaultPolicy>(octets, value);
+        }
+
+        // http://www.openmobilealliance.org/release/Browser_Protocol_Stack/V2_1-20110315-A/OMA-WAP-TS-WSP-V1_0-20110315-A.pdf,
+        // p81
+        template <typename PageShiftPred, typename NamePred, typename ValuePred>
+        static bool parseHeader(Octets &octets, PageShiftPred pageShiftPred, NamePred namePred, ValuePred valuePred)
+        {
+            HeaderName name = std::uint8_t(0);
+            if (!parseGeneric<HeaderNamePolicy>(octets, name)) {
+                return false;
+            }
+
+            // Distinguish between page shift and short integer
+            std::uint32_t nameUint = 0;
+            std::visit(
+                [&](auto const &n) {
+                    using n_t = utils::remove_cref_t<decltype(n)>;
+                    if constexpr (std::is_same_v<n_t, std::uint8_t>) {
+                        nameUint = n;
+                    }
+                },
+                name);
+            if (nameUint > 0) {
+                if (nameUint < PageShift) {
+                    pageShiftPred(static_cast<std::uint8_t>(nameUint));
+                    return true;
+                }
+                else if (nameUint == PageShift) {
+                    std::uint8_t octet = 0;
+                    if (!octets.next(octet)) {
+                        return false;
+                    }
+                    pageShiftPred(octet);
+                    return true;
+                }
+                else if (nameUint > TextMax && nameUint <= 0xFF) {
+                    name = std::uint8_t(nameUint & 0x7F); // decode short integer
+                }
+                else if (nameUint > 0xFF) {
+                    return false; // This should not happen
+                }
+            }
+
+            if (!namePred(std::move(name))) {
+                int foo;
+                return parseGeneric<IgnoreHeaderValuePolicy>(octets, foo);
+            }
+
+            HeaderValue value = std::uint8_t(0);
+            if (!parseGeneric<DefaultPolicy>(octets, value)) {
+                return false;
+            }
+
+            valuePred(std::move(value));
+            return true;
+        }
+
+        template <std::size_t N>
+        static bool parseHeaders(Octets &octets,
+                                 std::array<KnownHeader, N> const &headerNames,
+                                 std::array<HeaderValue, N> &headerValues)
+        {
+            std::array<bool, N> headersFound;
+            headersFound.fill(false);
+            std::size_t foundCount = 0;
+
+            while (!octets.empty()) {
+                std::size_t nameId      = N;
+                HeaderValue headerValue = std::uint8_t(0);
+
+                auto pageShiftPred = [&](std::uint8_t page) {
+                    // TODO: Support page shifts
```

Comment:
please don't leave TODO in the code - create a followup task in jira:
https://github.com/mudita/MuditaOS/blob/master/doc/MuditaCppCodingGuidelines.md#m-9-there-shall-be-no-comments-like-todo-fixme-etc

Reply:
Done